browser/ui/views/brave_help_bubble/brave_help_bubble_view.cc

```diff
@@ -0,0 +1,209 @@
+    tracked_element_->SetProperty(views::kElementIdentifierKey, id);
+  }
+
+  context_ = views::ElementTrackerViews::GetContextForView(tracked_element_);
+  CHECK(context_);
+
+  shown_subscription_ =
+      ui::ElementTracker::GetElementTracker()->AddElementShownCallback(
+          id, context_,
+          base::BindRepeating(&BraveHelpBubbleView::OnTrackedElementShown,
+                              base::Unretained(this)));
+  hidden_subscription_ =
+      ui::ElementTracker::GetElementTracker()->AddElementHiddenCallback(
+          id, context_,
+          base::BindRepeating(&BraveHelpBubbleView::OnTrackedElementHidden,
+                              base::Unretained(this)));
+}
+
+void BraveHelpBubbleView::AddedToWidget() {
+  InitElementTrackers();
+
+  if (tracked_element_ && tracked_element_->GetVisible()) {
+    UpdatePosition();
+    Show();
+  }
+}
+
+void BraveHelpBubbleView::UpdatePosition() {
+  auto tracked_element_origin =
+      tracked_element_->GetBoundsInScreen().CenterPoint();
+  tracked_element_origin.set_x(tracked_element_origin.x() - (kWidth / 2) - 1.f);
+  tracked_element_origin.set_y(tracked_element_origin.y() - kHeight - 15);
+
+  SetPosition({tracked_element_origin.x(), tracked_element_origin.y()});
+}
+
+void BraveHelpBubbleView::Show() {
+  if (!brave_help_bubble_delegate_) {
+    brave_help_bubble_delegate_ = new BraveHelpBubbleDelegate(this, text_);
+    brave_help_bubble_delegate_->AddObserver(this);
+  }
+
+  if (brave_help_bubble_delegate_) {
+    brave_help_bubble_delegate_->Show();
+  }
+
+  SetPaintToLayer();
+  layer()->SetFillsBoundsOpaquely(false);
+  SetSize({kWidth, kHeight});
+  SchedulePulsingAnimation(layer());
+  SetVisible(true);
+}
+
+void BraveHelpBubbleView::Hide() {
+  if (brave_help_bubble_delegate_) {
+    brave_help_bubble_delegate_->Hide();
+  }
+
+  SetVisible(false);
+}
+
+void BraveHelpBubbleView::OnBubbleClosing(Widget* widget) {
+  // In the destruction, we don't have to remove this from parent. Otherwise,
+  // DCHECK will fail as the ancestor view is iterating its children to destroy
+  // all descendant.
+  if (GetWidget()->IsClosed())
+    return;
+
+  // There's a possibility that someone is iterating over the parent's children,
+  // and we are removing the child view early. This can cause a DCHECK in
+  // view.cc, so we schedule removal of the child view via TaskRunner to avoid
+  // this issue.
+  base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
+      FROM_HERE,
+      base::BindOnce(
+          [](base::WeakPtr<views::View> view, View* this_element) {
+            if (!view)
+              return;
+            if (!view->parent() || !view->parent()->Contains(view.get()))
+              return;
+            view->parent()->RemoveChildView(this_element);
+          },
+          weak_factory_.GetWeakPtr(), this));
+
+  brave_help_bubble_delegate_ = nullptr;
+  base::SequencedTaskRunner::GetCurrentDefault()->DeleteSoon(FROM_HERE, this);
```

Comment:
Sorry, I know this part was what I suggested, but can we try like this?
The reason is..

* The view looks like it should be contained by a parent
* If we use RemoveChildViewT, we might be able to remove DeleteSoon() call below.

```suggestion
            DCHECK(view->parent());
            view->parent()->RemoveChildViewT(view.get());
          },
          weak_factory_.GetWeakPtr(), this));

  brave_help_bubble_delegate_ = nullptr;
```

Reply:
I don't understand how `DeleteSoon` might not be needed after calling `RemoveChildViewT`? From the comment looks like it will just remove the view from the tree and return a std::unique_ptr. 

I believe we have to take care of the memory management ourselves here?

Reply:
Looks like we can just leave the returned unique ptr here and once it's out of scope it will call it's dtor

https://chromium.googlesource.com/chromium/src/+/master/docs/ui/learn/bestpractices/ownership.md#use-view_removechildviewt